helpers/GenerateResourcesAndImage.ps1

```diff
@@ -266,16 +266,23 @@ Function GenerateResourcesAndImage {
                 $ResourceGroupExists = $false
             }
             else {
-                # Resource group already exists, ask the user what to do
-                $title = "Resource group '$ResourceGroupName' already exists"
-                $message = "Do you want to delete the resource group and all resources in it?"
+                # are we running in a non-interactive session?
+                # https://stackoverflow.com/questions/9738535/powershell-test-for-noninteractive-mode
+                if ([System.Console]::IsOutputRedirected -or ![Environment]::UserInteractive -or !!([Environment]::GetCommandLineArgs() | Where-Object { $_ -ilike '-noni*' })) {
+                    throw "Non-interactive mode, resource group '$ResourceGroupName' already exists, specify -Force to delete it."
```

Comment:
There is also 'ReuseResourceGroup' flag that is worth mentioning.

Reply:
makes sense